test/functional/test_framework/test_node.py

```diff
@@ -232,6 +232,15 @@ def wait_for_rpc_connection(self):
         poll_per_s = 4
         for _ in range(poll_per_s * self.rpc_timeout):
             if self.process.poll() is not None:
+                if self.process.returncode != 0:
+                    # Print any abrupt shutdown error, so it is visible from the console and the
+                    # user doesn't have to go node by node checking which one failed.
+                    with open(self.stderr.name, 'r', encoding='utf-8') as err_file:
+                        print("************************")
+                        print(f"'{self.datadir_path.name}' abruptly aborted with error:\n")
+                        print(''.join(line for line in err_file if line.strip() and line[0] != "*"))
+                        print("************************")
+
                 raise FailedToStartError(self._node_msg(
                     'bitcoind exited with status {} during initialization'.format(self.process.returncode)))
```

Comment:
The idea was to separate the stderr output from the exception being thrown here. Mainly to get a more readable print when a crash happens. The `FailedToStartError` exception message appears in-between two long stack traces.

But, either way is ok for me.

Reply:
Why might the output differ between CI and non-CI runs? The test framework redirect stderr to a file, and this file remains untouched regardless of the execution environment.

Also, the `"normal" test (not create_cache)` tense isn't clear for me. Does it refer to running an individual test locally?

Reply:
> > Why might the output differ between CI and non-CI runs?
> 
> I thought that the test_runner.py may be doing something with stderr, but it looks like it will get printed:
> 
> ```python
>                 print(BOLD[1] + 'stdout:\n' + BOLD[0] + stdout + '\n')
>                 print(BOLD[1] + 'stderr:\n' + BOLD[0] + stderr + '\n')
>                 if combined_logs_len and os.path.isdir(testdir):
>                     # Print the final `combinedlogslen` lines of the combined logs
> ```

In a first glance, that doesn't seems to print the nodes' stderr. That seems to be related to the job stderr (the test run in a subprocess). And those aren't connected.

>  > Also, the `"normal" test (not create_cache)` tense isn't clear for me. Does it refer to running an individual test locally?
> 
> Yes, I mean any test that is not `create_cache.py`.

Ok. The result is pretty much the same. Just the stack trace is a bit longer.

> > Mainly to get a more readable print when a crash happens. The `FailedToStartError` exception message appears in-between two long stack traces.
> 
> Looking at my code block above, it looks like the error message would be easier to spot in the output if it was part of the combined output, as opposed to the python stack trace, no?
> 
> You can use newlines to aid reading and break long blocks.

As far as I can see, we don't support combined output on single test cases. So, that would make us lose the crash cause on individual test runs?